specification/cognitiveservices/Language.AnalyzeText/models/common.models.tsp

```diff
@@ -0,0 +1,788 @@
+
+@doc("The status of the task at the mentioned last update time.")
+@clientName("CurrentState", "csharp")
+@lroStatus
+union State {
+  string,
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "The EnumMember named 'notStarted' should have a documentation or description, please use decorator @doc to add it."
+  notStarted: "notStarted",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "The EnumMember named 'running' should have a documentation or description, please use decorator @doc to add it."
+  running: "running",
+
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "The EnumMember named 'succeeded' should have a documentation or description, please use decorator @doc to add it."
+  @lroSucceeded
+  succeeded: "succeeded",
+
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "The EnumMember named 'partiallyCompleted' should have a documentation or description, please use decorator @doc to add it."
+  partiallyCompleted: "partiallyCompleted",
+
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "The EnumMember named 'failed' should have a documentation or description, please use decorator @doc to add it."
+  @lroFailed
+  failed: "failed",
+
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "The EnumMember named 'cancelled' should have a documentation or description, please use decorator @doc to add it."
+  @lroCanceled
+  cancelled: "cancelled",
+
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "The EnumMember named 'cancelling' should have a documentation or description, please use decorator @doc to add it."
+  cancelling: "cancelling",
+}
+
+@doc("Base task object.")
+model TaskIdentifier {
+  @doc("task name")
+  taskName?: string;
+}
+
+@clientName("TaskContent", "csharp")
+@doc("Base parameters object for a text analysis task.")
+model TaskParameters {
+  @doc("logging opt out")
+  loggingOptOut?: boolean = false;
+}
+
+@clientName("PreBuiltTaskContent", "csharp")
+@doc("Parameters object for a text analysis task using pre-built models.")
+model PreBuiltTaskParameters is TaskParameters {
+  @doc("model version")
+  modelVersion?: string = "latest";
+}
+
+alias ShowStatsQueryParameter = QueryParametersTrait<{
+  @doc("(Optional) if set to true, response will contain request and document level statistics.")
+  @query
+  showStats?: boolean;
+}>;
+
+alias AnalyzeTextJobQueryParameters = QueryParametersTrait<{
+  @doc("(Optional) if set to true, response will contain request and document level statistics.")
+  @query
+  showStats?: boolean;
+
+  @doc("The maximum number of resources to return from the collection.")
+  @query
+  top?: int32;
+
+  @doc("An offset into the collection of the first resource to be returned.")
+  @query
+  skip?: int32;
+}>;
+
+@doc("Query parameters for the AnalyzeText operation.")
+model AnalyzeTextQueryParameters {
+  ...Azure.Core.Foundations.ApiVersionParameter;
```

Comment:
Given the recommendations I made for the operation definitions, you can now remove this from this model since you get the Api version param through the templates